cpp/src/arrow/testing/random.cc

```diff
@@ -42,6 +42,7 @@
 #include "arrow/util/decimal.h"
 #include "arrow/util/key_value_metadata.h"
 #include "arrow/util/logging.h"
+#include "arrow/util/optional.h"
```

Comment:
nit: I don't see any uses of optional/nullopt in this file?

Reply:
Right, that was for a previous attempt. Will fix.